src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
I'm not sure what you're saying here. This code is not correct as-is, because multiple registers are being used, but you are not calling `genConsumeReg` for each one.